homeassistant/components/onewire/onewire_entities.py

```diff
@@ -83,24 +83,22 @@ class OneWireProxyEntity(OneWireBaseEntity):
 
     def __init__(
         self,
-        name: str,
-        device_file: str,
-        entity_type: str,
-        entity_name: str,
+        device_id: str,
+        device_name: str,
         device_info: Dict[str, Any],
-        default_disabled: bool,
+        entity_path: str,
+        entity_specs: Dict[str, Any],
         owproxy: protocol._Proxy,
-        unique_id: str,
     ):
         """Initialize the sensor."""
         super().__init__(
-            name,
-            device_file,
-            entity_type,
-            entity_name,
-            device_info,
-            default_disabled,
-            unique_id,
+            name=device_name,
+            device_file=entity_path,
+            entity_type=entity_specs["type"],
+            entity_name=entity_specs["name"],
+            device_info=device_info,
+            default_disabled=entity_specs.get("default_disabled", False),
+            unique_id=f"/{device_id}/{entity_specs['path']}",
```

Comment:
Your commit is saying "Move shared logic into OneWireBaseEntity".

But you made changes on `OneWireProxyEntity`, is it normal ?
I think it should be on `OneWireBaseEntity`.

Also, you can remove `entity_type` , `entity_name`, `default_disabled`, and so `unique_id` from `OneWireBaseEntity` constructor, and pass directly `entity_specs` to it. But I don't know if possible, I did not check all the integration code.

Reply:
It was a bad commit message => I should have written it as "Move shared logic into OneWireProxyEntity"

I prefer not to make any changes to OneWireBaseEntity as that is shared with OWFS (deprecated - we don't have any tests on it) and with OWDirect implementations.

Reply:
@epenet OWFS was deprecated in https://github.com/home-assistant/core/pull/42041. Not sure about the removal policy, but I think it should be removed in a release or two. Then the changes to `OneWireBaseEntity` can be done.

Reply:
Yeah - chicken and egg issue.
I think it's better to sort out OneWireProxyEntity now, and then do OneWireBaseEntity when we remove OWFS.

OWFS was deprecated in 0.117, but right now OWFS users can't migrate to OWProxy because of missing coupler support (#43599) which depends in this PR!

Reply:
OK, good with that.